src/v3/src/transformer/selectAuthenticator/transformSelectAuthenticatorUnlockVerify.ts

```diff
@@ -38,25 +49,133 @@ export const transformSelectAuthenticatorUnlockVerify: IdxStepTransformer = ({
+    type: 'Title',
+    options: {
+      content: loc('oie.select.authenticators.verify.title', 'login'),
+    },
+  };
+
+  const verifySubtitle: DescriptionElement = {
+    type: 'Description',
+    contentType: 'subtitle',
+    options: {
+      content: loc('oie.select.authenticators.verify.subtitle', 'login'),
+    },
+  };
+
   const identifier = getUIElementWithName('identifier', uischema.elements) as FieldElement;
 
-  const authenticatorListElement: AuthenticatorButtonListElement = {
+  if (identifier) {
+    // add username/identifier from config if provided
+    if (username) {
+      data.identifier = username;
+    } else if (typeof identifier.options.inputMeta.value === 'string') {
+      data.identifier = identifier.options.inputMeta.value;
+    // TODO: OKTA-508744 - to use rememberMe in features once Default values are added widgetProps.
+    // (i.e. rememberMe is default = true in v2)
+    } else if (typeof features?.rememberMe === 'undefined' || features?.rememberMe === true) {
+      const usernameCookie = getUsernameCookie();
+      data.identifier = usernameCookie;
+    }
+  }
+
+  const authenticatorList: AuthenticatorButtonListElement = {
     type: 'AuthenticatorButtonList',
     options: { buttons: authenticatorButtons, dataSe: 'authenticator-verify-list' },
   };
 
-  uischema.elements = [
-    title,
+  const identifyWithUsernameLayout: UISchemaLayout = {
+    type: UISchemaLayoutType.VERTICAL,
+    elements: [],
+  };
+
+  // TODO: Refactor transformer once we have backend support for this Stepper flow - OKTA-657627
+  const nextButton: StepperButtonElement = {
+    type: 'StepperButton',
+    label: loc('oform.next', 'login'),
+    options: {
+      type: ButtonType.SUBMIT,
+      variant: 'primary',
+      nextStepIndex: (widgetContext) => {
+        const { setMessage, data: updatedData } = widgetContext;
+
+        if (updatedData.identifier && typeof updatedData.identifier === 'string') {
+          setMessage(undefined);
+          identifierContainer.options.identifier = (updatedData.identifier as string);
+
+          // If the user only has one authenticator that they can use to verify, add the AutoSubmit
+          // element to skip rendering the authenticator list and proceed with a challenge request
+          if (authenticatorButtons.length === 1) {
+            const authenticatorOptions = authenticatorButtons[0].options;
+            const autoSubmit: AutoSubmitElement = {
+              type: 'AutoSubmit',
+              options: {
+                step: stepName,
+                actionParams: authenticatorOptions.actionParams,
+                includeData: authenticatorOptions.includeData,
+                includeImmutableData: authenticatorOptions.includeImmutableData,
+              },
+            };
+            identifyWithUsernameLayout.elements.push(autoSubmit);
+            // Keep users on first layout because AutoSubmit will trigger view change
+            return 0;
```

Comment:
Just curious, for this part, is it possible to allow the user to advance to the next view but just conditionally add the autoSubmit element to the next view's elements array? This way, we are not modifying the elements array here.

Reply:
So I was originally doing that but it resulted in some sub-optimal UX where the user is taken to the authenticator list for a second or so and then it jumps to the authenticator challenge page. I chatted with Alex and he said that the ideal UX would be for the Next button to switch to a loading state with a spinner (hence why I had to add support for a `Spinner` in the `StepperButton`)

Reply:
Ahh, got it, that makes sense. 👍